webui/react/src/pages/Admin/UserManagement.tsx

```diff
@@ -323,57 +376,77 @@ const UserManagement: React.FC = () => {
       },
     ];
     return rbacEnabled ? columns.filter((c) => c.dataIndex !== 'isAdmin') : columns;
-  }, [
-    fetchUsers,
-    filterIcon,
-    groups,
-    info.userManagementEnabled,
-    nameFilterSearch,
-    rbacEnabled,
-    settings,
-  ]);
-
-  const table = useMemo(() => {
-    return settings ? (
-      <InteractiveTable<DetailedUser, UserManagementSettings>
-        columns={columns}
-        containerRef={pageRef}
-        dataSource={filteredUsers}
-        interactiveColumns={false}
-        loading={Loadable.isLoading(loadableUsers)}
-        rowClassName={defaultRowClassName({ clickable: false })}
-        rowKey="id"
-        settings={{
-          ...settings,
-          columns: DEFAULT_COLUMNS,
-          columnWidths: DEFAULT_COLUMNS.map((col: UserColumnName) => DEFAULT_COLUMN_WIDTHS[col]),
-        }}
-        showSorterTooltip={false}
-        size="small"
-        updateSettings={updateSettings}
-      />
-    ) : (
-      <SkeletonTable columns={columns.length} />
-    );
-  }, [filteredUsers, loadableUsers, settings, columns, updateSettings]);
+  }, [fetchUsers, groups, info.userManagementEnabled, rbacEnabled, settings]);
 
   return (
     <>
-      <Section
-        className={css.usersTable}
-        options={
-          <Space>
-            <Button
-              aria-label={CREATE_USER_LABEL}
-              disabled={!info.userManagementEnabled || !canModifyUsers}
-              onClick={CreateUserModal.open}>
-              {CREATE_USER}
-            </Button>
-            {settings.name && <Button onClick={handleNameSearchReset}>{'Clear Filter'}</Button>}
-          </Space>
-        }
-        title={USER_TITLE}>
-        {table}
+      <Section className={css.usersTable}>
+        <div className={css.actionBar}>
+          <Columns>
+            <Column>
+              <Columns>
+                {/* input is uncontrolled to prevent settings overwriting during composition */}
+                <Input
+                  defaultValue={settings.name}
+                  prefix={<Icon color="cancel" decorative name="search" />}
+                  onChange={handleNameSearchApply}
+                />
+                <Select
+                  allowClear
+                  options={roleOptions}
+                  placeholder="All roles"
+                  searchable={false}
+                  value={settings.roleFilter}
+                  onChange={handleRoleFilterApply}
+                />
+                <Select
```

Comment:
<img width="156" alt="image" src="https://github.com/determined-ai/determined/assets/109113616/60d94560-3d2f-4b6b-b578-6ebefd961ee0">

prob width should be set to avoid text truncation?